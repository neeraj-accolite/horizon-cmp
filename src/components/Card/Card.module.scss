.card-container {
    transition: box-shadow 0.3s ease, border-color 0.3s ease;
  }
  
  .card {
    border: 1px solid grey;
    border-radius: 12px;
    overflow: hidden;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 16px;
    text-align: center;
  
    &:hover {
      box-shadow: 0 4px 16px rgba(0, 0, 0, 0.2);
      border-color: black;
    }
  
}