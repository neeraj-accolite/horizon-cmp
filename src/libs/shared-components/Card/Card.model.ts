export interface CardProps {
  children: React.ReactNode;
  style?: React.CSSProperties;
}
