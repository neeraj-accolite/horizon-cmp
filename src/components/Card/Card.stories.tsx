import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { Card, CardProps } from './Card';

export default {
  title: 'Components/Card',
  component: Card,
  parameters: { layout: 'centered' },
} as Meta;

const Template: StoryFn<CardProps> = (args) => <Card {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Generic Card',
  children: <div>Child Component</div>,
  width: '400px',
  height: '300px',
};
