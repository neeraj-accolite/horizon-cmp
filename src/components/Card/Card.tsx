import React from 'react';
import styles from './Card.module.scss';
import { CardProps } from './Card.model';

export const Card: React.FC<CardProps> = ({
  title,
  children,
  width,
  height,
}) => {
  return (
    <div className={`${styles['card-container']} transition-shadow`}>
      <div
        className={`${styles.card} flex flex-col items-center p-4 text-center`}
        style={{
          width: width,
          height: height,
        }}
      >
        <h2 className="text-xl mb-2">{title}</h2>
        <div className="flex-1 w-full">{children}</div>
      </div>
    </div>
  );
};

export type { CardProps };