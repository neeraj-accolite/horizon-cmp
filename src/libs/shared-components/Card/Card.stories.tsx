import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { Card } from './Card';
import { CardProps } from './Card.model';

export default {
  title: 'Components/Card',
  component: Card,
  parameters: { layout: 'centered' },
} as Meta;

const Template: StoryFn<CardProps> = (args) => <Card {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: <div>Default Child Component</div>,
  style: {
    borderColor: '#D9D9D9',
    backgroundColor: '#F5F6F6',
    borderRadius: '8px',
    padding: '1rem',
  },
};

export const CustomStyle = Template.bind({});
CustomStyle.args = {
  children: <div>Custom Styled Child Component</div>,
  style: {
    borderColor: '#FF5733',
    backgroundColor: '#C70039',
    borderRadius: '16px',
    padding: '2rem',
    color: '#FFFFFF',
  },
};
