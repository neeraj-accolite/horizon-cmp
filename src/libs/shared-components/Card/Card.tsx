import React from 'react';
import { CardProps } from './Card.model';

const Card: React.FC<CardProps> = ({
  children,
  style = {},
  className = '',
}) => {
  return (
    <div style={{ ...style }} className={className}>
      {children}
    </div>
  );
};

export default Card;
