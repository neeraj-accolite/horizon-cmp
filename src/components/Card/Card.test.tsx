import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Card } from './Card';
import { CardProps } from './Card.model';

describe('Card Component', () => {
  const defaultProps: CardProps = {
    title: 'Test Card',
    children: <div>Test Child</div>,
    width: '400px',
    height: '300px',
  };

  it('renders the card with the correct title', () => {
    render(<Card {...defaultProps} />);
    expect(screen.getByText('Test Card')).toBeInTheDocument();
  });

  it('renders the card with the correct children', () => {
    render(<Card {...defaultProps} />);
    expect(screen.getByText('Test Child')).toBeInTheDocument();
  });

  it('applies hover styles correctly', () => {
    render(<Card {...defaultProps} />);
    const cardElement =
      screen.getByText('Test Card').parentElement?.parentElement;
    expect(cardElement).toHaveClass('transition-shadow');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Card {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
