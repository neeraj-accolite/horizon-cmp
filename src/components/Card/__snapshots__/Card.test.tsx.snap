// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card-container transition-shadow"
  >
    <div
      class="card flex flex-col items-center p-4 text-center"
      style="width: 400px; height: 300px;"
    >
      <h2
        class="mb-2 text-xl"
      >
        Test Card
      </h2>
      <div
        class="w-full flex-1"
      >
        <div>
          Test Child
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
